Doc/library/string.rst

```diff
@@ -476,20 +476,32 @@ with the floating point presentation types listed below (except
 ``'n'`` and ``None``). When doing so, :func:`float` is used to convert the
 integer to a floating point number before formatting.
 
-The available presentation types for floating point and decimal values are:
+The available presentation types for :class:`float` and
+:class:`~decimal.Decimal` values are:
 
    +---------+----------------------------------------------------------+
    | Type    | Meaning                                                  |
    +=========+==========================================================+
-   | ``'e'`` | Exponent notation. Prints the number in scientific       |
-   |         | notation using the letter 'e' to indicate the exponent.  |
-   |         | The default precision is ``6``.                          |
+   | ``'e'`` | Scientific "E" notation. For a given precision ``p >=    |
+   |         | 0``, formats the number in scientific notation with the  |
+   |         | letter 'e' separating the coefficient from the exponent. |
+   |         | The coefficient has one digit before and ``p`` digits    |
+   |         | after the decimal point, for a total of ``p + 1``        |
+   |         | significant digits. With no precision given, uses a      |
+   |         | precision of ``6`` digits after the decimal point for    |
+   |         | :class:`float`, and shows all coefficient digits         |
+   |         | for :class:`~decimal.Decimal`.                           |
```

Comment:
Should you mention here that if `p == 0`, the decimal point is omitted unless `#` is used?

Also: is it generally understood that for Decimal, "all coefficient digits" includes the trailing zeros? Again, I can see people reporting it as an issue until they're told that the trailing zeros are significant.

Reply:
Yep, my original draft for this text had an "(including trailing zeros)"; I can put it back. I took it out partly because the text seemed to be getting too long, and partly out of a fear that it might cause more confusion rather than less and lead to a need to explain exactly what's meant by "trailing zeros". And this doesn't seem like the right place for that explanation.

So ... not sure.

Reply:
And yes, I'll update for the missing decimal point, both here and for the `f` text.

Reply:
You're probably right about leaving out the "trailing zeros". There's a lot here already.

Reply:
> if `p == 0`, the decimal point is omitted unless `#` is used

Argh. The reality is messy. Decimal apparently still doesn't support the `#` alternate format. (There ought to be an issue open about this somewhere.)

There's a paragraph below the table about the inclusion or not of the decimal point. I'll see if I can update that to reflect reality.

Reply:
Related: https://bugs.python.org/7094, https://bugs.python.org/13838